src/ivectorbin/merge-vads.cc

```diff
@@ -0,0 +1,163 @@
+      std::vector<std::string> fields;
+      SplitStringToVector(line.substr(start, end - start + 1),
+         " \t\n\r", true, &fields);
+      if (fields.size() != 3) {
+        KALDI_ERR << "Bad line. Expected three fields, got: "
+                  << line;
+      }
+      int32 label1 = std::atoi(fields[0].c_str()),
+            label2 = std::atoi(fields[1].c_str()),
+            result_label = std::atoi(fields[2].c_str());
+      (*map)[std::pair<int32, int32>(label1, label2)] = result_label;
+    }
+  }
+}
+
+}
+
+int main(int argc, char *argv[]) {
+  using namespace kaldi;
+  typedef kaldi::int32 int32;
+  try {
+    const char *usage =
+      "This program merges two archives of per-frame weights representing\n"
+      "voice activity decisions.  By default, the program assumes that the\n"
+      "input vectors consist of floats that are 0.0 if a frame is judged\n"
+      "as nonspeech and 1.0 if it is considered speech.  The default\n"
+      "behavior produces a frame-level decision of 1.0 if both input frames\n"
+      "are 1.0, and 0.0 otherwise.  Additional classes (e.g., 2.0 for music)\n"
+      "can be handled using the \"map\" option.\n"
+      "\n"
+      "Usage: merge-vads [options] <vad-rspecifier-1> <vad-rspecifier-2>\n"
+      "    <vad-wspecifier>\n"
+      "e.g.: merge-vads [options] scp:vad_energy.scp scp:vad_gmm.scp\n"
+      "    ark:vad.ark\n"
+      "See also: compute-vad-from-frame-likes, compute-vad, ali-to-post,\n"
+      "post-to-weights\n";
+
+    ParseOptions po(usage);
+    std::string map_rxfilename;
+    po.Register("map", &map_rxfilename, "This table specifies a mapping "
+      "between the labels of the frame-level decisions in the first and "
+      "second input archives to the integer output label.");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() != 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    unordered_map<std::pair<int32, int32>, int32, PairHasher<int32> > map;
+    PrepareMap(map_rxfilename, &map);
+    SequentialBaseFloatVectorReader first_vad_reader(po.GetArg(1));
+    RandomAccessBaseFloatVectorReader second_vad_reader(po.GetArg(2));
+    BaseFloatVectorWriter vad_writer(po.GetArg(3));
+
+    int32 num_done = 0, num_err = 0;
+    for (;!first_vad_reader.Done(); first_vad_reader.Next()) {
+      std::string utt = first_vad_reader.Key();
+      Vector<BaseFloat> vad1(first_vad_reader.Value());
+      if (!second_vad_reader.HasKey(utt)) {
+        KALDI_WARN << "No vector for utterance " << utt;
+        num_err++;
+        continue;
+      }
+      Vector<BaseFloat> vad2(second_vad_reader.Value(utt));
+      if (vad1.Dim() != vad2.Dim()) {
+        KALDI_WARN << "VAD length mismatch for utterance " << utt;
+        num_err++;
+        continue;
+      }
+      Vector<BaseFloat> vad_result(vad1.Dim());
+      for (int32 i = 0; i < vad1.Dim(); i++) {
+        std::pair<int32, int32> key(static_cast<int32>(vad1(i)),
+          static_cast<int32>(vad2(i)));
```

Comment:
it might make sense to add a warning if you detect non-integer inputs. can be done later though.